### PR TITLE
fix(startup): make recovery mode work without --headless

### LINKED ISSUE #23294: `nvim -r` hangs (or returns nothing)
### Problem

Running `nvim -r` hangs (after clearing terminal screen). On another machine, `nvim -r` returns instantly with exit status 0, but `nvim -r -V` hangs like on the first machine. It does not matter if I have another nvim instance open or not.

### Steps to reproduce

nvim -r

### Expected behavior

List swap files with recovery information.

### Neovim version (nvim -v)

v0.9.0

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Archlinux up-to-date

### Terminal name/version

foot version: 1.14.0 +pgo +ime +graphemes -assertions

### $TERM environment variable

foot-extra

### Installation

official arch repos

### PR DESCRIPTION
**Pull request by zeertzjq**
Fix #23294

### CODE REVIEW

File: test/functional/core/startup_spec.lua [hunk 1]
```diff
@@ -460,6 +461,18 @@ describe('startup', function()
     ]])
   end)
 
+  it('-r works without --headless in PTY #23294', function()
+    skip(is_os('win'))
```
Comment (justinmk) [inline test/functional/core/startup_spec.lua:465]: Why doesn't this work on WIndows?

Comment (zeertzjq) [reply test/functional/core/startup_spec.lua:465]: It hangs even with `--headless`. Maybe a bug in PTY or libuv code.

Comment (zeertzjq) [reply test/functional/core/startup_spec.lua:465]: Oh, it seems to be `jobwait()` causing the problem here.

Comment (justinmk) [reply test/functional/core/startup_spec.lua:465]: Ah right, jobwait() has some quirks on Windows IIRC.

**Pull request merged**